camel/models/gemini_model.py

```diff
@@ -0,0 +1,185 @@
+        )
+        response.resolve()
+        return self.to_openai_response(response)
+
+    def check_model_config(self):
+        r"""Check whether the model configuration contains any
+        unexpected arguments to Gemini API.
+
+        Raises:
+            ValueError: If the model configuration dictionary contains any
+                unexpected arguments to OpenAI API.
+        """
+        if self.model_config_dict is not None:
+            for param in self.model_config_dict:
+                if param not in Gemini_API_PARAMS:
+                    raise ValueError(
+                        f"Unexpected argument `{param}` is "
+                        "input into Gemini model backend."
+                    )
+
+    @property
+    def stream(self) -> bool:
+        r"""Returns whether the model is in stream mode,
+            which sends partial results each time.
+        Returns:
+            bool: Whether the model is in stream mode.
+        """
+        return self.model_config_dict.get('stream', False)
+
+    def to_gemini_req(self, req):
+        r"""Converts the request from the OpenAI API format to
+            the Gemini API request format.
+
+        Args:
+            req: The request object from the OpenAI API.
+        Returns:
+            converted_messages: A list of messages formatted for Gemini API.
+        """
+        converted_messages = []
+        for message in req:
+            role = message.get('role')
+            if role == 'assistant':
+                role = 'model'
+            else:
+                role = 'user'
+            if 'content' in message:
+                converted_message = {
+                    "role": role,
+                    "parts": message.get("content"),
+                }
+                converted_messages.append(converted_message)
+            else:
+                converted_messages.append(message)
+        return converted_messages
+
+    def to_openai_response(self, res):
+        r"""Converts the response from the Gemini API to the OpenAI API
+        response format.
+        Args:
+            res: The response object returned by the Gemini API
+        Returns:
+            openai_res: A ChatCompletion object formatted for the OpenAI API.
+        """
+        import time
+        import uuid
+
+        openai_res = ChatCompletion(
+            id=f"chatcmpl-{uuid.uuid4().hex!s}",
+            object="chat.completion",
+            created=int(time.time()),
+            model="gemini",
+            choices=[],
+        )
+        for i, candidate in enumerate(res.candidates):
+            content = ""
+            if candidate.content and len(candidate.content.parts) > 0:
+                content = candidate.content.parts[0].text
+            choice = Choice(
+                index=i,
+                message=ChatCompletionMessage(
+                    role="assistant", content=content
+                ),
+                finish_reason='stop',
```

Comment:
it's may not good to hard code the role and finish reason, if these information is not provided from gemini, maybe we can just set this as None